### ISSUE TITLE
Add more detailed explanation of how to use remark-lint programmatically

### ISSUE THREAD

**Issue by EvgenyOrekhov** [wooorm/remark-lint#131]
I spent several hours trying to figure out how to use remark 7.0.0 and remark-lint 6.0.0 programmatically, but I still don't understand how.
The documentation is inconsistent and scattered over several repos, which makes it hard to follow.

I ended up with the following piece of code that produces unexpected results:

```js
const remark = require("remark");
const remarkLint = require("remark-lint");
const consistentPreset = require("remark-preset-lint-consistent");
const recommendedPreset = require("remark-preset-lint-recommended");
const finalNewlineRule = require("remark-lint-final-newline");

remark()
    .use(remarkLint)
    .use(consistentPreset)
    .use(recommendedPreset)
    .use(finalNewlineRule, false)
    .process("# a\n# b #")
    .then((result) => console.log(result.messages));
```

It produces the following warnings:

```
Headings should use atx
Missing newline character at end of file
Headings should use atx
Missing newline character at end of file
Headings should use atx
Missing newline character at end of file
```

1. Why are they repeated 3 times each?
2. Why is the "Missing newline character at end of file" warning present? I switched it off by invoking `.use(finalNewlineRule, false)`.

This example on RunKit: https://runkit.com/57e78997c63d1e1400b5bd9f/58b5ce6baf786e00135b3141/branches/master

**Comment by wooorm** [comment]
Terribly sorry about that, first of all the update (it’s quite the process, to update hundreds of projects), and lastly this bug.
I think I’ve fixed it, but I want Travis to finish first (that might take some time as S3 is failing).

**Comment by wooorm** [comment]
The bug fix is released as 6.1.1!
